python/ray/train/callbacks/print.py

```diff
@@ -0,0 +1,53 @@
+import json
+from typing import Dict, List
+
+from ray.train.callbacks import TrainingCallback
+
+
+class PrintCallback(TrainingCallback):
+    """A callback that prints training results to STDOUT.
+
+    Example:
+
+        >>> from ray import train
+        >>> from ray.train import Trainer
+        >>> from ray.train.callbacks import PrintCallback
+
+        >>> def train_func():
+        ...    for i in range(2):
+        ...        train.report(worker_idx=train.world_rank(), epoch=i)
```

Comment:
nit: Preference on `>>>` vs. `..code-block` formatting? Personally prefer to just be able to copy/paste the example and run it.

Reply:
My reasoning was that the `>>>` formatting would be easier to show what's actually being printed out. If we do `..code-block` then we would have to show the stdout as a comment. But good point on being able to copy paste. I don't really have a strong preference here, so am happy to change it to code block.